src/client/auth/actions.js

```diff
@@ -33,14 +33,23 @@ function validateForm(fields) {
 }
 
 function validateCredentials(fields) {
-  // Simulate lengthy async action.
   return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      if (fields.password === 'pass1')
+
+    var xhr = new XMLHttpRequest();
```

Comment:
We are using [isomorphic-fetch](https://github.com/matthew-andrews/isomorphic-fetch) with our Este app and it works very well. Even in IE9. The interface is much better than XHR of course and it works well both on server and on client.